packages/Discretization/src/DTK_Interpolation_def.hpp

```diff
@@ -0,0 +1,669 @@
+    : public DirectSerializationTraits<Ordinal, DataTransferKit::Point>
+{
+};
+}
+
+namespace DataTransferKit
+{
+template <typename DeviceType>
+Interpolation<DeviceType>::Interpolation(
+    Teuchos::RCP<const Teuchos::Comm<int>> comm,
+    Kokkos::View<DTK_CellTopology *, DeviceType> cell_topologies_view,
+    Kokkos::View<unsigned int *, DeviceType> cells,
+    Kokkos::View<double **, DeviceType> nodes_coordinates,
+    Kokkos::View<double **, DeviceType> points_coordinates )
+    : _apply_allowed( false )
+    , _comm( comm )
+    , _distributor( comm )
+    , _cell_topologies_view( cell_topologies_view )
+    , _bounding_boxes( "bounding_boxes", _cell_topologies_view.extent( 0 ) )
+    , _bounding_box_to_cell( "bounding_box_to_cell",
+                             _cell_topologies_view.extent( 0 ), DTK_N_TOPO )
+{
+
+    _cell_topologies[0] = shards::getCellTopologyData<shards::Triangle<3>>();
+    _cell_topologies[1] = shards::getCellTopologyData<shards::Triangle<6>>();
+    _cell_topologies[2] =
+        shards::getCellTopologyData<shards::Quadrilateral<4>>();
+    _cell_topologies[3] =
+        shards::getCellTopologyData<shards::Quadrilateral<9>>();
+    _cell_topologies[4] = shards::getCellTopologyData<shards::Tetrahedron<4>>();
+    _cell_topologies[5] =
+        shards::getCellTopologyData<shards::Tetrahedron<10>>();
+    _cell_topologies[6] =
+        shards::getCellTopologyData<shards::Tetrahedron<11>>();
+    _cell_topologies[7] = shards::getCellTopologyData<shards::Hexahedron<8>>();
+    _cell_topologies[8] = shards::getCellTopologyData<shards::Hexahedron<20>>();
+    _cell_topologies[9] = shards::getCellTopologyData<shards::Hexahedron<27>>();
+    _cell_topologies[10] = shards::getCellTopologyData<shards::Pyramid<5>>();
+    _cell_topologies[11] = shards::getCellTopologyData<shards::Pyramid<13>>();
+    _cell_topologies[12] = shards::getCellTopologyData<shards::Wedge<6>>();
+    _cell_topologies[13] = shards::getCellTopologyData<shards::Wedge<15>>();
+    _cell_topologies[14] = shards::getCellTopologyData<shards::Wedge<18>>();
+
+    for ( unsigned int topo_id = 0; topo_id < DTK_N_TOPO; ++topo_id )
+        _n_cells_per_topo[topo_id] = 0;
+
+    findReferencePoints( cells, nodes_coordinates, points_coordinates );
+}
+
+template <typename DeviceType>
+Interpolation<DeviceType>::Interpolation(
+    Teuchos::RCP<const Teuchos::Comm<int>> comm,
+    Kokkos::View<DTK_CellTopology *, DeviceType> cell_topologies_view,
+    Kokkos::View<unsigned int *, DeviceType> cells,
+    Kokkos::View<double **, DeviceType> nodes_coordinates,
+    Kokkos::View<double **, DeviceType> points_coordinates,
+    Kokkos::View<LocalOrdinal *, DeviceType> object_dof_ids )
+    : Interpolation<DeviceType>( comm, cell_topologies_view, cells,
+                                 nodes_coordinates, points_coordinates )
+{
+    _apply_allowed = true;
+
+    // This function does exactly the same thing as convertMesh but for
+    // object_dof_ids
```

Comment:
After talking with @sslattery, we have come to the conclusion that we should get ride of the `shards` objects. However, I would like to do in another PR because it means changes in `PointInCell` and the PR is already large enough so I would prefer not change more files.